docs/vaadin-grid-columns.adoc

```diff
@@ -168,6 +168,11 @@ endif::web[]
 The end users can reorder non-frozen columns by dragging the header cells.
 By default, column reordering is disabled, but can be enabled by setting the [propertyname]#columnReorderingAllowed# property to true.
 
+[NOTE]
+Column reordering is only supported when the data source is based on objects bound to the column configuration using the `name` attribute.
```

Comment:
It should work as well with array datasources as long as the source returns correctly ordered data after column reorder